sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5770,6 +5905,46 @@ bool _pi_queue::useCopyEngine(bool PreferCopyEngine) const {
          (!isInOrderQueue() || UseCopyEngineForInOrderQueue);
 }
 
+// Wait on all operations in flight on this Queue.
+// If using immediate commandlists add barriers to all commandlists associated
+// with the Queue. An alternative approach would be to wait on all Events
+// associated with the in-flight operations. For standard commandlists sync the
+// L0 queues directly.
+pi_result _pi_queue::synchronize() {
```

Comment:
add a comment that the Queue lock should be held by the caller of this function.

Reply:
OK.